app/models/completed_resource.rb

```diff
@@ -0,0 +1,16 @@
+# == Schema Information
+#
+# Table name: completed_resources
+#
+#  id          :bigint           not null, primary key
+#  resource_id :bigint           not null
+#  user_id     :bigint           not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+class CompletedResource < ApplicationRecord
+  belongs_to :user
+  belongs_to :resource
+
```

Comment:
I think we are adding model validation for "presence" of user and resource as well as validation table (that is OK!).